### DISCUSSION TITLE
How to build a power flow card

### DISCUSSION THREAD

**Discussion by EmAnzi3** [wlcrs/huawei_solar#90] [General]
Hello,
First of all many thanks for this helpful integration, it works like a charm in the HA Energy tab.
However, I was searching for a solution to replicate in HA a power flow card like the one of FusionSolar app.
My setup is a bit articulated since I have 2 inverters and 3 batteries and I'm a bit lost with parameter to be used and summed together.
Thanks in advance for your kind support.

![Immagine 2022-07-17 102540](https://user-images.githubusercontent.com/27973961/179402345-62a31d35-562f-4f6b-99d9-cf283ff98146.png)

**Comment by bdkacz** [comment]
You can create template sensors that will sum the values from several sensors and connect them as source of data  to the card

**Comment by JacoboSanchez** [comment]
I would point you to [this question](https://github.com/reptilex/tesla-style-solar-power-card/issues/95) in the tesla style solar power card but that was already opened by you on last December. So you already know that lines of power going from one entity to another is not the data the inverter provides. It is up to you to figure out how the power flows in each moment and do the maths for your particular installation topology (which we do not even know how it is and it is not for sure a standard one).

Energy tab works with energy so it does not really care about what it is happening in any instant but the sum of all energy imported/exported

The sample I provided you in the referenced question was done by drawing the topology in a paper, writting real values on different scenarios (exporting energy, importing, charging battery...etc) and trying to figure the formula for each flow line
